cryptomator: 1.5.15 -> 1.6.7

bachp opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Update to latest version and fix segault when trying to mount a tresor.


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

bachp committed:
cryptomator: 1.5.15 -> 1.6.5

- packer is not needed anymore, remove it
- use nix built jffi instead of bundeled one to avoid segfault

bachp committed:
jffi: init at 1.3.9

peterwilli commented:
Hi @bachp 

I tried your PR, but during build it keeps spitting out the following error over and over again in a loop:
![image](https://user-images.githubusercontent.com/1212814/159119651-58083970-81e7-41bc-99c4-fbe2fa0cf381.png)

Do you have the same issue? In my PR, I fixed it with this: https://github.com/NixOS/nixpkgs/pull/150402/files#diff-a31db4c783608e1de3f54a95ba3726d10fe8c6351ada3c129ca8f7ac95ef0c1dR33

Since you don't have this fix, I start to wonder, is this an issue solely on my end? I can't imagine being a fully functional OS.

My nix-info:

```
system: "x86_64-linux", multi-user?: yes, version: nix-env (Nix) 2.3.16, channels(root): "nixos-21.11.336538.64fc73bd74f, nixos-unstable-22.05pre360796.73ad5f9e147", nixpkgs: /nix/var/nix/profiles/per-user/root/channels/nixos
```

bachp commented:
@peterwilli I got the error with OpenJFX 15. That's why I submitted the update to OpenJFX 17.
Could it be that you are using a Java version from before [OpenJFX 17](https://github.com/NixOS/nixpkgs/pull/155746) was merged?

peterwilli commented:
> @peterwilli I got the error with OpenJFX 15. That's why I submitted the update to OpenJFX 17. Could it be that you are using a Java version from before [OpenJFX 17](https://github.com/NixOS/nixpkgs/pull/155746) was merged?

Oh! Yeah. Most likely, I dropped this in my own NixPkgs fork! I can't believe that it even built without showing an error about not having OpenJFX 17! Thanks!

SuperSandro2000 commented on pkgs/development/libraries/java/jffi/default.nix:
```diff
@@ -0,0 +1,47 @@
+  buildPhase = ''
+    # The pkg-config script in the build.xml doesn't work propery
+    # set the lib path manually to work around this.
+    export LIBFFI_LIBS="${libffi}/lib/libffi.so"
```
Shouldn't we fix it rather than setting always the env?

bachp replied:
@SuperSandro2000 The error I get when not setting this is:

```
 last 10 log lines:
              >     [javac] warning: [options] bootstrap class path not set in conjunction with -source 8
              >     [javac] 1 warning
              >
              > -build-native-library:
              >      [exec] make: *** No rule to make target '-L/nix/store/20ix3np9v02ph8fwb2v41r5mzlfg8f73-libffi-3.4.2/lib/../lib64', needed by '/build/source/build/jni/libjffi-1.2.so'.  Stop.
              >
              > BUILD FAILED
              > /build/source/build.xml:372: exec returned: 2

```

So I'm not sure if the `libffi.pc` file is wrong or if `jffi` uses it in an incompatible way.

bachp committed:
cryptomator: 1.6.5 -> 1.6.7

Disable maven tests as they try to mount and load kernel modules,
which is not somthing we want during a build.

bachp commented:
Updated to 1.6.7

mannp commented:
@bachp many thanks for the update 👍🏻 wanted to report my version number is showing as 'Snapshot' and I have lost my icon in gnome....as well as the gnome application name showing as 'java' rather than cryptomator.

I am not sure how to check the GUI version outside of the 'snapshot' to know which version I am actually using.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.16.18, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0`
 - channels(root): `"home-manager, nixos, nixos-unstable-small"`
 - channels(local): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```
